server/monitor-types/system-service.js

```diff
@@ -0,0 +1,110 @@
+     * Detects OS and dispatches to the appropriate check method.
+     * @param {object} monitor The monitor object containing monitor.system_service_name.
+     * @param {object} heartbeat The heartbeat object to update.
+     * @returns {Promise<void>} Resolves when check is complete.
+     */
+    async check(monitor, heartbeat) {
+        // Use the new variable name 'system_service_name' to match the monitor type change
+        const serviceName = monitor.system_service_name;
+
+        // Basic sanitization.
+        // We do not allow spaces to ensure users use the "Service Name" (wuauserv) and not "Display Name".
+        if (!serviceName || !/^[a-zA-Z0-9._\-@]+$/.test(serviceName)) {
+            heartbeat.status = DOWN;
+            heartbeat.msg = "Invalid service name. Please use the internal Service Name (no spaces).";
+            throw new Error(heartbeat.msg);
+        }
+
+        if (process.platform === "win32") {
+            return this.checkWindows(serviceName, heartbeat);
+        } else {
+            return this.checkLinux(serviceName, heartbeat);
+        }
+    }
+
+    /**
+     * Linux Check (Systemd)
+     * @param {string} serviceName The name of the service to check.
+     * @param {object} heartbeat The heartbeat object.
+     * @returns {Promise<void>} Resolves on success, rejects on error.
+     */
+    async checkLinux(serviceName, heartbeat) {
+        return new Promise((resolve, reject) => {
+            // Linter requires spaces inside array brackets: [ "arg1", "arg2" ]
+            execFile("systemctl", [ "is-active", serviceName ], (error, stdout, stderr) => {
+                // Combine output and truncate to ~200 chars to prevent DB bloat
+                let output = (stderr || stdout || "").toString().trim();
+                if (output.length > 200) {
+                    output = output.substring(0, 200) + "...";
+                }
+
+                if (error) {
+                    heartbeat.msg = output || `Service '${serviceName}' is not running.`;
+                    reject(new Error(heartbeat.msg));
+                    return;
+                }
+
+                heartbeat.status = UP;
+                heartbeat.msg = `Service '${serviceName}' is running.`;
+                resolve();
+            });
+        });
+    }
+
+    /**
+     * Windows Check (PowerShell)
+     * @param {string} serviceName The name of the service to check.
+     * @param {object} heartbeat The heartbeat object.
+     * @returns {Promise<void>} Resolves on success, rejects on error.
+     */
+    async checkWindows(serviceName, heartbeat) {
+        return new Promise((resolve, reject) => {
+            const cmd = "powershell";
+            // -NoProfile: Faster startup, -NonInteractive: No prompts
+            const args = [
+                "-NoProfile",
+                "-NonInteractive",
+                "-Command",
+                `(Get-Service -Name "${serviceName}").Status`
```

Comment:
Looks like command injection?

Reply:
Space character is actually allowed. 

Example:
```
(Get-Service -Name "ROG Live Service").Status
```

However, I know it is not easy to be handled, I think it is OK to not allow space if we cannot find a better solution.

<img width="300"  alt="image" src="https://github.com/user-attachments/assets/35d8bbef-017e-408c-85d1-e5ff9d68369b" />